cmd/mt-store-cat/main.go

```diff
@@ -0,0 +1,152 @@
+	cassandraCaPath           = flag.String("cassandra-ca-path", "/etc/raintank/ca.pem", "cassandra CA certificate path when using SSL")
+	cassandraHostVerification = flag.Bool("cassandra-host-verification", true, "host (hostname and server cert) verification when using SSL")
+
+	cassandraAuth     = flag.Bool("cassandra-auth", false, "enable cassandra authentication")
+	cassandraUsername = flag.String("cassandra-username", "cassandra", "username for authentication")
+	cassandraPassword = flag.String("cassandra-password", "cassandra", "password for authentication")
+
+	// our own flags
+	from = flag.String("from", "-24h", "get data from (inclusive)")
+	to   = flag.String("to", "now", "get data until (exclusive)")
+	mdp  = flag.Int("mdp", 0, "max data points to return")
+)
+
+func main() {
+	flag.Usage = func() {
+		fmt.Println(os.Args[0])
+		fmt.Println("Usage:")
+		fmt.Printf("	mt-store-cat [flags] id <metric-id>\n")
+		fmt.Printf("	mt-store-cat [flags] query <org-id> <graphite query> (not supported yet)\n")
+		fmt.Println("Flags:")
+		flag.PrintDefaults()
+	}
+	flag.Parse()
+
+	if flag.NArg() < 2 {
+		flag.Usage()
+		os.Exit(-1)
+	}
+
+	mode := flag.Arg(0)
+	var id string
+	// var query string
+	// var org int
+
+	switch mode {
+	case "id":
+		id = flag.Arg(1)
+	case "query":
+		//		if flag.NArg() < 3 {
+		//			flag.Usage()
+		//			os.Exit(-1)
+		//		}
+		//		org64, err := strconv.ParseInt(flag.Arg(2), 10, 32)
+		//		if err != nil {
+		//			flag.Usage()
+		//			os.Exit(-1)
+		//		}
+		//		org = int(org64)
+		//		query = flag.Arg(3)
+		panic("sorry, queries not supported yet")
+	default:
+		flag.Usage()
+		os.Exit(-1)
+	}
+
+	now := time.Now()
+
+	defaultFrom := uint32(now.Add(-time.Duration(24) * time.Hour).Unix())
+	defaultTo := uint32(now.Add(time.Duration(1) * time.Second).Unix())
+
+	fromUnix, err := dur.ParseTSpec(*from, now, defaultFrom)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	toUnix, err := dur.ParseTSpec(*to, now, defaultTo)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Only try and parse the conf file if it exists
+	path := ""
+	if _, err := os.Stat(*confFile); err == nil {
+		path = *confFile
+	}
+	conf, err := globalconf.NewWithOptions(&globalconf.Options{
+		Filename:  path,
+		EnvPrefix: "MT_",
+	})
+	if err != nil {
+		log.Fatal(4, "error with configuration file: %s", err)
+		os.Exit(1)
+	}
+
+	conf.ParseAll()
+
+	if *showVersion {
+		fmt.Printf("mt-store-cat (built with %s, git hash %s)\n", runtime.Version(), GitHash)
+		return
+	}
+
+	fmt.Println("looking for id", id)
+	stats, err := helper.New(false, "", "standard", "mt-store-cat", "default")
+
+	store, err := mdata.NewCassandraStore(stats, *cassandraAddrs, *cassandraKeyspace, *cassandraConsistency, *cassandraCaPath, *cassandraUsername, *cassandraPassword, *cassandraHostSelectionPolicy, *cassandraTimeout, *cassandraReadConcurrency, 0, *cassandraReadQueueSize, 0, *cassandraRetries, *cqlProtocolVersion, *cassandraSSL, *cassandraAuth, *cassandraHostVerification)
```

Comment:
you can't set writers to 0, as this is used for setting the number of connections to Cassandra.  It will be hard to query Cassandra if you have 0 connections.